src/Utility.php

```diff
@@ -120,6 +120,28 @@ public static function get_value( array $array, string $key ) {
         return null;
     }
 
+    /**
+     * Escapes a string value but prevents numeric values from being slashed. 
+     * Numeric values will be casted to the correct type.
+     *
+     * @param mixed $string The string to escape.
+     * @param string $meta_type MySQL data type.
+     * @return mixed depending on the meta_type parameter and input string 
+     * the return value will be either string (or array of strings), int or float.
+     */
+    public static function escape_value_by_meta_type( $string, $meta_type = 'CHAR' ):mixed{
+
+        // Possible numeric MySQL data type values for WP meta query are SIGNED, UNSIGNED, NUMERIC and DECIMAL.
+        // For NUMERIC and DECIMAL declaration also data precicion and scale can be specified.
+        // https://dev.mysql.com/doc/refman/8.0/en/numeric-type-syntax.html
```

Comment:
You could change this comment to a comment block 'cause they are preferred with multiline comments.

```
/**
 * Possible numeric...
 */
 ```